ledger/stateproofverificationtracker.go

```diff
@@ -169,41 +167,27 @@ func (spt *stateProofVerificationTracker) close() {
 	}
 }
 func (spt *stateProofVerificationTracker) LookupVerificationContext(stateProofLastAttestedRound basics.Round) (*ledgercore.StateProofVerificationContext, error) {
-	if lstlookup := spt.retrieveFromCache(stateProofLastAttestedRound); lstlookup != nil {
-		return lstlookup, nil
+	spt.mu.RLock()
+	defer spt.mu.RUnlock()
+
+	// If cached - retrieve
+	if spt.lastLookedUpVerificationContext.LastAttestedRound == stateProofLastAttestedRound &&
+		!spt.lastLookedUpVerificationContext.MsgIsZero() {
+		return &spt.lastLookedUpVerificationContext, nil
 	}
 
-	verificationContext, err := spt.lookUpVerificationContext(stateProofLastAttestedRound)
+	verificationContext, err := spt.lookupVerificationContext(stateProofLastAttestedRound)
 	if err != nil {
 		return nil, err
 	}
 
-	spt.mu.Lock()
 	spt.lastLookedUpVerificationContext = *verificationContext
```

Comment:
we taking RLock above but have write op here

Reply:
now I'm not sure I like this change. This change will cause the lookup function to take a write lock the entire time. Before it only took it when updating the cache 
What do you think?